online-shop/src/app/auth.guard.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate,  Route, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+import { AuthService} from "./auth.service";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivate {
+  constructor(private authService: AuthService, private router: Router) {}
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean | UrlTree {
+    const url: string = state.url;
+
+    return this.check(url, route);
+  }
+
+  check(url: string, route: ActivatedRouteSnapshot): boolean | UrlTree {
+    if (this.authService.isLoggedIn) {
+      console.log(route.data['roles'])
```

Comment:
**suggestion:** remove console.log from the application after you finish with your debugging